patches/server/0085-Implement-SpawnerPreSpawnEvent.patch

```diff
@@ -0,0 +1,39 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: mechoriet <kevinworm92@gmail.com>
+Date: Mon, 19 Dec 2022 23:48:44 +0100
+Subject: [PATCH] Implement SpawnerPreSpawnEvent with check to see if there are
+ listeners for it so we can limit the amount of events fired
+
+
+diff --git a/src/main/java/net/minecraft/server/MobSpawnerAbstract.java b/src/main/java/net/minecraft/server/MobSpawnerAbstract.java
+index e54f7edbd1d834391f6760ea7b82cf6af4ddf994..fad53e6e97520579f3f9b8ba2f1c7fbed6e6f047 100644
+--- a/src/main/java/net/minecraft/server/MobSpawnerAbstract.java
++++ b/src/main/java/net/minecraft/server/MobSpawnerAbstract.java
+@@ -5,6 +5,7 @@ import java.util.Iterator;
+ import java.util.List;
+ 
+ // CraftBukkit start
++import net.techcable.tacospigot.event.entity.SpawnerPreSpawnEvent;
+ import org.bukkit.craftbukkit.event.CraftEventFactory;
+ import org.bukkit.event.entity.CreatureSpawnEvent;
+ import org.bukkit.event.entity.SpawnerSpawnEvent;
+@@ -91,6 +92,19 @@ public abstract class MobSpawnerAbstract {
+                 boolean flag = false;
+ 
+                 for (int i = 0; i < this.spawnCount; ++i) {
++                    // PandaSpigot start - add check to avoid firing if non listeners
++                    if(SpawnerPreSpawnEvent.getHandlerList().getRegisteredListeners().length > 0) {
++                        // TacoSpigot start
```

Comment:
Taco 😆

Reply:
bruh

Reply:
@uRyanxD cause its a 1.8.8 patch from taco https://github.com/TacoSpigot/TacoSpigot/blob/version/1.8.8/PaperSpigot-Server-Patches/0022-Implement-SpawnerPreSpawnEvent.patch

and added only the start and end to the change I made to limit broadcasting that event if there are listeners for it 
since it cancelable and spawning code is generally hotter since it ran very often so kind of  a waste to construct a class every time it tries to spawn things